src/sankey/index.js

```diff
@@ -1,153 +1,198 @@
 import React, {Component} from 'react';
 import PropTypes from 'prop-types';
-import {sankey} from 'd3-sankey-align';
-
+import {
+  sankey,
+  sankeyLinkHorizontal,
+  sankeyLeft,
+  sankeyRight,
+  sankeyCenter,
+  sankeyJustify
+} from 'd3-sankey';
+import XYPlot from 'plot/xy-plot';
+
+import VerticalRectSeries from 'plot/series/vertical-rect-series';
+import LabelSeries from 'plot/series/label-series';
 import Voronoi from 'plot/voronoi';
 import {DISCRETE_COLOR_RANGE} from 'theme';
 
+import SankeyLink from './sankey-link';
 const NOOP = f => f;
 
-const DEFAULT_LINK_COLOR = DISCRETE_COLOR_RANGE[1];
-const DEFAULT_LINK_OPACITY = 0.7;
-const DEFAULT_NODE_COLOR = DISCRETE_COLOR_RANGE[0];
-const DEFAULT_NODE_OPACITY = 1;
+const ALIGNMENTS = {
+  justify: sankeyJustify,
+  center: sankeyCenter,
+  left: sankeyLeft,
+  right: sankeyRight
+};
 
 class Sankey extends Component {
-
-  static defaultProps = {
-    align: 'justify',
-    className: '',
-    hasVoronoi: false,
-    hideLabels: false,
-    layout: 50,
-    margin: 20,
-    nodePadding: 10,
-    nodeWidth: 10,
-    onBlur: NOOP,
-    onClick: NOOP,
-    onHover: NOOP
-  }
-
-  static propTypes = {
-    align: PropTypes.oneOf(['justify', 'left', 'right', 'center']),
-    className: PropTypes.string,
-    hasVoronoi: PropTypes.bool,
-    height: PropTypes.number.isRequired,
-    hideLabels: PropTypes.bool,
-    layout: PropTypes.number,
-    links: PropTypes.arrayOf(PropTypes.shape({
-      source: PropTypes.oneOfType([
-        PropTypes.number,
-        PropTypes.object
-      ]).isRequired,
-      target: PropTypes.oneOfType([
-        PropTypes.number,
-        PropTypes.object
-      ]).isRequired
-    })).isRequired,
-    margin: PropTypes.number,
-    nodePadding: PropTypes.number,
-    nodes: PropTypes.arrayOf(PropTypes.object).isRequired,
-    nodeWidth: PropTypes.number,
-    onBlur: PropTypes.func,
-    onClick: PropTypes.func,
-    onHover: PropTypes.func,
-    width: PropTypes.number.isRequired
-  }
-
   render() {
 
     const {
       align,
+      animation,
       className,
       hasVoronoi,
       height,
       hideLabels,
       layout,
       links,
+      linkOpacity,
       margin,
       nodePadding,
       nodes,
       nodeWidth,
-      onBlur,
-      onClick,
-      onHover,
+      onValueClick,
+      onValueMouseOver,
+      onValueMouseOut,
+      style,
       width
     } = this.props;
+    const nodesCopy = [...new Array(nodes.length)].map((e, i) => ({...nodes[i]}));
```

Comment:
Not sure why we need to force copying the nodes and links? Shouldn't that be handled by the app itself rather than the component?

Reply:
I think the idea is that we want the users to have to do as little configuration as possible, if we require them to make copies just for us, that will most likely be sort of error prone?